File: src/Adaptive.ReactiveTrader.Web/IReactiveTrader.ts
 interface IReactiveTrader {
     tradeRepository: ITradeRepository;
     referenceDataRepository: IReferenceDataRepository;
     priceLatencyRecorder: IPriceLatencyRecorder;
     connectionStatusStream: Rx.Observable<ConnectionInfo>;
     initialize(username: string, servers: string[]): void;
     dispose(): void;
 } 

File: src/Adaptive.ReactiveTrader.Client.Domain/IReactiveTrader.cs
using System;
using Adaptive.ReactiveTrader.Client.Domain.Instrumentation;
using Adaptive.ReactiveTrader.Client.Domain.Repositories;
using Adaptive.ReactiveTrader.Shared.Logging;

namespace Adaptive.ReactiveTrader.Client.Domain
{
    public interface IReactiveTrader
    {
        IReferenceDataRepository ReferenceData { get; }
        ITradeRepository TradeRepository { get; }
        IObservable<ConnectionInfo> ConnectionStatusStream { get; }
        IPriceLatencyRecorder PriceLatencyRecorder { get; }
        IControlRepository Control { get; }
        void Initialize(string username, string[] servers, ILoggerFactory loggerFactory = null, string authToken = null);
    }
}